### ISSUE TITLE
Demos does not work on Chromium 97.0.4692.99 Arch Linux

### ISSUE THREAD

**Issue by gwpl** [sixtyfpsui/sixtyfps#896]
Demos does not work on Chromium 97.0.4692.99 Arch Linux.

(However they work in Firefox...)

**Comment by ogoffart** [comment]
They work for me on the same version of Chromium.
Could you describe what is not working?

**Comment by gwpl** [comment]
On Firefox everything works fine.

On Chromium in a place where should be wasm sixtyfps demo apps, it's just white.

**Comment by gwpl** [comment]
E.g. https://sixtyfps.io/releases/0.1.6/demos/slide_puzzle/

when I open Developer Console I see:

```
slide_puzzle_bg.wasm:0x1bbbec Uncaught (in promise) RuntimeError: unreachable
    at slide_puzzle_bg.wasm:0x1bbbec
    at slide_puzzle_bg.wasm:0x1c5dc3
    at slide_puzzle_bg.wasm:0x1c2da0
    at slide_puzzle_bg.wasm:0x3fd80
    at slide_puzzle_bg.wasm:0x90f33
    at slide_puzzle_bg.wasm:0xd4f12
    at init (slide_puzzle.js:1268:10)
$func2168	@	slide_puzzle_bg.wasm:0x1bbbec
$func2697	@	slide_puzzle_bg.wasm:0x1c5dc3
$func2496	@	slide_puzzle_bg.wasm:0x1c2da0
$func270	@	slide_puzzle_bg.wasm:0x3fd80
$func309	@	slide_puzzle_bg.wasm:0x90f33
$main	@	slide_puzzle_bg.wasm:0xd4f12
init	@	slide_puzzle.js:1268
```

**Comment by ogoffart** [comment]
Interresting. The exact same works for me.
And i'm also using an up-to-date Archlinux

![Screenshot_20220129_132157](https://user-images.githubusercontent.com/959326/151660837-0a837a2e-6565-45f3-9c36-7634289fccb5.png)
.

**Comment by gwpl** [comment]
Yes, I know. It works for me on Firefox on Arch Linux, and on Chromium on Ubuntu. But not on Arch Linux.

Based on stack trace , are you able to identify control flow ?

Otherwise, if you would have some debugging build with tracing information dumped to console, I am more then happy to try it out and post here results from console.

**Comment by tronical** [comment]
Perhaps this is a webgl related problem. What graphics card do you have? Does this demo work for you in Chrome? https://mdn.github.io/webgl-examples/tutorial/sample3/

What does https://webglreport.com/ report for your setup?

**Comment by gwpl** [comment]
![Screenshot_20220129_183232](https://user-images.githubusercontent.com/221403/151671139-03468983-c0f4-4174-b756-d203980f6d36.png)

Bingo! I am getting "Unable to initialize WebGL. Your browser or machine may not support it".

Maybe good to add to sixtyfps demos/examples/templates error handling code to display error message (webgl or any other failure)?

**Comment by tronical** [comment]
I agree, we should show an alert in that case.

**Comment by ogoffart** [comment]
The main() of our demo could return a Result<> instead of unwrap'ing the Slint result.
and the entry point of wasm could react to it and do an alert.

**Comment by tronical** [comment]
That would be very nice, indeed! I think for that to work, we need to provide an implementation of `Into<JsValue>` for `PlatformError` (at least my local attempt at doing this bails out with this requirement).

We could perhaps have a weak dependency on wasm-bindgen in i-slint-core and then implement it there, but that would imply that wasm-bindgen becomes a public dependency in our API, I guess.

**Comment by fivlao** [comment]
I think we close this issue

**Comment by ogoffart** [comment]
This was done for the slide puzzle in #7917 , but it should be done for every of out wasm examples.
